hugolib/page.go

```diff
@@ -467,7 +468,8 @@ func (p *Page) LinkTitle() string {
 }
 
 func (p *Page) ShouldBuild() bool {
```

Comment:
This method has gotten complex enough to deserve a refactor.

I would pull out a function (as in not a method) with:

* buildFuture
* buildExpired
* buildDrafts
* publishDate
* expireDate

I.e. no dependency on Page nor Viper, then create a table driven test for that function.